### PR TITLE
[1.0.0.1] Add initial helm resources for Choreo Connect Deployment

### PR DESCRIPTION
**Pull request by renuka-fernando**
## Purpose
Resolves https://github.com/wso2/kubernetes-microgateway/issues/22
$subject

Find sample ReadMe: https://github.com/renuka-fernando/kubernetes-microgateway/tree/choreo-connect-100

### CODE REVIEW

File: helm/choreo-connect/README.md [hunk 1]
```diff
@@ -0,0 +1,465 @@
...
+```
+git clone https://github.com/wso2/kubernetes-microgateway.git
+```
+
+##### 2. Deploy Helm chart for Choreo Connect deployment.
+
+ Helm version 2
+
+ ```
+ helm install --name <RELEASE_NAME> <HELM_HOME>/choreo-connect --version 1.0.0-1 --namespace <NAMESPACE>
```
Comment (chirangaalwis) [inline helm/choreo-connect/README.md:92]:
```suggestion
 helm install --name <RELEASE_NAME> <HELM_HOME>/choreo-connect --namespace <NAMESPACE>
```

File: helm/choreo-connect/README.md [hunk 2]
```diff
@@ -0,0 +1,465 @@
...
+ ```
+
+ Helm version 3
+
+```
+helm install <RELEASE_NAME> <HELM_HOME>/choreo-connect --version 1.0.0-1 --namespace <NAMESPACE> --create-namespace
```
Comment (chirangaalwis) [inline helm/choreo-connect/README.md:98]:
```suggestion
helm install <RELEASE_NAME> <HELM_HOME>/choreo-connect --namespace <NAMESPACE> --create-namespace
```

File: helm/choreo-connect/README.md [hunk 3]
```diff
@@ -0,0 +1,465 @@
...
+
+The above steps will deploy the deployment pattern using WSO2 product Docker images available at DockerHub.
+
+If you are using WSO2 product Docker images available from WSO2 Private Docker Registry,
+please provide your WSO2 Subscription credentials via input values (using `--set` argument). 
+
+Please see the following example.
+
+```
+ helm install --name <RELEASE_NAME> <HELM_HOME>/choreo-connect --version 1.0.0-1 --namespace <NAMESPACE> --set wso2.subscription.username=<SUBSCRIPTION_USERNAME> --set wso2.subscription.password=<SUBSCRIPTION_PASSWORD>
```
Comment (chirangaalwis) [inline helm/choreo-connect/README.md:109]:
```suggestion
 helm install --name <RELEASE_NAME> <HELM_HOME>/choreo-connect --namespace <NAMESPACE> --set wso2.subscription.username=<SUBSCRIPTION_USERNAME> --set wso2.subscription.password=<SUBSCRIPTION_PASSWORD>
```

File: helm/choreo-connect/auth.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
+	"auths": {
+		"reg.id": {
+			"username": "docker.wso2.com.username",
+			"password": "docker.wso2.com.password",
+			"email": "docker.wso2.com.email",
+			"auth": "docker.wso2.com.auth"
+		}
+	}
+}
```
Comment (chirangaalwis) [inline helm/choreo-connect/auth.json:10]: @renuka-fernando please add a new EOF line.

**Review by chirangaalwis — CHANGES_REQUESTED** [review]

File: helm/choreo-connect/templates/enforcer-grpc-probe-script-configmap.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    # -----------------------------------------------------------------------
+
+    ENFORCER_PORT="${ENFORCER_PORT:-8081}"
+    grpc_health_probe -addr "localhost:${ENFORCER_PORT}" \
+        -tls \
+        -tls-ca-cert "${ENFORCER_PUBLIC_CERT_PATH}" \
+        -tls-client-cert "${ENFORCER_PUBLIC_CERT_PATH}" \
+        -tls-client-key "${ENFORCER_PRIVATE_KEY_PATH}" \
+        -tls-server-name {{ .Values.wso2.deployment.gatewayRuntime.enforcer.security.sslHostname }} \
+        -connect-timeout=3s
```
Comment (chirangaalwis) [inline helm/choreo-connect/templates/enforcer-grpc-probe-script-configmap.yaml:48]: @renuka-fernando please add a new EOF line.

**Review by chirangaalwis — CHANGES_REQUESTED** [review]

**Review by pubudu538 — APPROVED** [review]

File: CHANGELOG.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
+# Changelog
+
+All notable changes to Kubernetes and Helm resources for Choreo Connect version `1.0.x` in each resource release,
+will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
+
+## [v1.0.0.1] - 2022-01-21
```
Comment (chirangaalwis) [inline CHANGELOG.md:8]:
```suggestion
## [v1.0.0.1] - 2022-01-24
```

**Review by chirangaalwis — CHANGES_REQUESTED** [review]

**Review by chirangaalwis — APPROVED** [review]

**Pull request merged**